src/main/java/org/akip/domain/TaskDefinition.java

```diff
@@ -0,0 +1,292 @@
+package org.akip.domain;
+
+import org.apache.tools.ant.Task;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.annotations.Type;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Entity
+@Table(name = "task_definition")
+@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
+public class TaskDefinition implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
+    @SequenceGenerator(name = "sequenceGenerator", sequenceName = "akip_hibernate_sequence")
+    private Long id;
+
+    @Column(name = "task_id")
+    private String taskId;
+
+    @Column(name = "name")
+    private String name;
+
+    @Lob
+    @Type(type = "org.hibernate.type.TextType")
+    @Column(name = "documentation")
+    private String documentation;
+
+    @Column(name = "assignee")
+    private String assignee;
+
+    @Column(name = "candidate_users")
+    private String candidateUsers;
+
+    @Column(name = "candidate_groups")
+    private String candidateGroups;
+
+    @Column(name = "priority")
+    private Integer priority;
+
+    @Column(name = "form_builder")
+    private String formBuilder;
+
+    @Column(name = "form_version")
+    private String formVersion;
+
+    @Lob
+    @Type(type = "org.hibernate.type.TextType")
+    @Column(name = "form_schema")
+    private String formSchema;
+
+    @Lob
+    @Type(type = "org.hibernate.type.TextType")
+    @Column(name = "props")
+    private String props;
+
+    @Lob
+    @Type(type = "org.hibernate.type.TextType")
+    @Column(name = "form_fields")
+    private String formFields;
+
+    @Column(name = "bpmn_process_definition_id")
+    private String bpmnProcessDefinitionId;
```

Comment:
How about the relationship with ProcessDefinition?

Reply:
I think it looks good too, I'll just need to change it in some places that are already using this property

Reply:
In fact, there is a problem with adding this relationship, I think, if you add the relationship, it becomes more complicated to add a TaskDefinition before deploying bpmn for the first time, because the processDefinition will not exist yet

Reply:
You can use natural id annotation and cascade none from TaskDefinition -> ProcessDefinition relation. See how I create a TaskInstance before committing a ProcessInstance. It's the same problem.